### PR TITLE
fix: color restrictions issue

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A-I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] N/A-I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: apps/web/components/eventtype/EventAdvancedTab.tsx [hunk 1]
```diff
@@ -607,13 +607,13 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupProps,
...
+                    };
+                    formMethods.setValue("eventTypeColor", newVal, { shouldDirty: true });
                     if (checkWCAGContrastColor("#101010", value)) {
-                      const newVal = {
-                        ...eventTypeColorState,
-                        darkEventTypeColor: value,
-                      };
                       setDarkModeError(false);
-                      formMethods.setValue("eventTypeColor", newVal, { shouldDirty: true });
                       setEventTypeColorState(newVal);
```
Comment (Amit91848) [inline apps/web/components/eventtype/EventAdvancedTab.tsx:617]:
```suggestion
```
Shouldn't this also be outside the if-else block?

File: apps/web/components/eventtype/EventAdvancedTab.tsx [hunk 2]
```diff
@@ -607,13 +607,13 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupProps,
                 <ColorPicker
                   defaultValue={eventTypeColorState.darkEventTypeColor}
                   onChange={(value) => {
+                    const newVal = {
+                      ...eventTypeColorState,
+                      darkEventTypeColor: value,
+                    };
+                    formMethods.setValue("eventTypeColor", newVal, { shouldDirty: true });
```
Comment (Amit91848) [inline apps/web/components/eventtype/EventAdvancedTab.tsx:614]:
```suggestion
                    formMethods.setValue("eventTypeColor", newVal, { shouldDirty: true });
                    setEventTypeColorState(newVal);
```

**Review by Amit91848 — CHANGES_REQUESTED** [review]
Hey @anikdhabal, I guess you missed moving a setState outside if-else block, otherwise changes look good

File: apps/web/components/eventtype/EventAdvancedTab.tsx (hunk 1 above)
Comment (anikdhabal) [reply apps/web/components/eventtype/EventAdvancedTab.tsx]: Yeah, I missed it

**Review by Amit91848 — APPROVED** [review]
LGTM!!

**Pull request merged**